maven-extension/src/main/java/io/opentelemetry/maven/OtelExecutionListener.java

```diff
@@ -200,8 +226,16 @@ public void mojoStarted(ExecutionEvent executionEvent) {
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_ID, mojoExecution.getExecutionId())
             .setAttribute(
                 MavenOtelSemanticAttributes.MAVEN_EXECUTION_LIFECYCLE_PHASE,
-                mojoExecution.getLifecyclePhase())
-            .startSpan();
+                mojoExecution.getLifecyclePhase());
+    //  enrich spans with MojoGoalExecutionHandler
```

Comment:
Thanks @cyrille-leclerc, looks good!

One question, as we're finding the first only, which I agree with, do we need to perform any validation earlier when retrieving the services to ensure there isn't a duplicate service handling the same goal and plugin combination?

Reply:
done, please check the new code.

Reply:
Thanks for the changes!